x/bank/keeper.go

```diff
@@ -73,20 +69,119 @@ func (ck CoinKeeper) SendCoins(ctx sdk.Context, fromAddr sdk.Address, toAddr sdk
 }
 
 // InputOutputCoins handles a list of inputs and outputs
-func (ck CoinKeeper) InputOutputCoins(ctx sdk.Context, inputs []Input, outputs []Output) sdk.Error {
+func inputOutputCoins(ctx sdk.Context, am sdk.AccountMapper, inputs []Input, outputs []Output) sdk.Error {
```

Comment:
Ditto on above - if an error is returned, the caller *must* revert state changes.